### ISSUE TITLE
PCC Drop in rtdetr-r18vd, PCC=-0.010627585533299516

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#3447]
### Describe the bug

- `rtdetr-r18vd(red model)` variant having `PCC=-0.010627585533299516` - [feb24_rtdetr_r18.log](https://github.com/user-attachments/files/25524374/feb24_rtdetr_r18.log)
- Model returns 14 outputs. Among them, init_reference_points exhibits a significantly low PCC compared to the others.
- Since logits themselves already have a low PCC, the focus should be on identifying the earliest point where PCC drops below 0.99.
- Below experiments confirms that PCC drop first originates from the [topk](https://github.com/huggingface/transformers/blob/8cb5963cc22174954e7dca2c0a3320b7dc2f4edc/src/transformers/models/rt_detr/modeling_rt_detr.py#L1734C29-L1734C33) op
- Exp 1 : return tensors before topk - `PCC=0.9995429296979129` - [feb24_rt_detr_r18_e1.log](https://github.com/user-attachments/files/25524461/feb24_rt_detr_r18_e1.log), [modeling_rt_detr.py](https://github.com/user-attachments/files/25524460/modeling_rt_detr.py)
- Exp 2 : return tensors after topk - `PCC=-0.011199128810294728` - [feb24_rt_detr_r18_e2.log](https://github.com/user-attachments/files/25524467/feb24_rt_detr_r18_e2.log), [modeling_rt_detr.py](https://github.com/user-attachments/files/25524468/modeling_rt_detr.py)
- Saved the inputs to topk, and passed to topk sanity - `PCC=0.5048749658552738` - [feb24_topk_s.log](https://github.com/user-attachments/files/25524689/feb24_topk_s.log)
- Exp 1 proves the input to topk between CPU and device is slightly different (PCC 0.99, not 1.0). Exp 2 shows topk amplifies that small difference. The sanity test can't reproduce the same drop because it feeds the exact same tensor to both CPU and device, eliminating the upstream drift. The only drop comes from the topk itself.

| #  | Output Field                    | PCC      |
|----|--------------------------------|----------|
| 0  | logits                         | 0.24729  |
| 1  | pred_boxes                     | 0.04683  |
| 2  | last_hidden_state             | 0.32339  |
| 3  | intermediate_hidden_states    | 0.32816  |
| 4  | intermediate_logits           | 0.24114  |
| 5  | intermediate_reference_points | 0.05961  |
| 6  | encoder_last_hidden_state[0]  | 0.99978  |
| 7  | encoder_last_hidden_state[1]  | 0.99983  |
| 8  | encoder_last_hidden_state[2]  | 0.99987  |
| 9  | init_reference_points         | -0.01063 |
| 10 | enc_topk_logits               | 0.04426  |
| 11 | enc_topk_bboxes               | 0.00545  |
| 12 | enc_outputs_class             | 0.99963  |
| 13 | enc_outputs_coord_logits      | 1.00000  |

### Repro

```
git checkout kkannan/feb24_topk_pcc_drop_rt_detr
git submodule update --init --recursive
git lfs pull --include '*.pt'

# model run 
pytest tests/runner/test_models.py::test_all_models_torch[rt_detr/pytorch-R18vd-single_device-inference] -svv

# sanity run 
pytest tests/torch/ops/test_topk_rt_detr.py -svv 

# emitpy run 
python examples/pytorch/codegen/python/topk.py
cd topk
./run 
```

### TT-METAL 

- Replicated the sanity's drop in tt-metal using TTNN repro - [feb24_topk_org.log](https://github.com/user-attachments/files/25526067/feb24_topk_org.log)
- More specifically, drop is from indices returned by `ttnn.sort` op. so Raised a ticket for sort op- https://github.com/tenstorrent/tt-metal/issues/38462

### Logs

- [feb24_topk_emitpy_gen.log](https://github.com/user-attachments/files/25524703/feb24_topk_emitpy_gen.log)
- [feb24_topk_emitpy_run.log](https://github.com/user-attachments/files/25524702/feb24_topk_emitpy_run.log)
- [topk.zip](https://github.com/user-attachments/files/25524739/topk.zip)
